apps/aehttp/src/aehttp_app.erl

```diff
@@ -47,28 +46,27 @@ stop(_State) ->
 %%====================================================================
 %% Internal functions
 %%====================================================================
-start_swagger_external() ->
-    Port = get_external_port(),
-    ListenAddress = get_external_listen_address(),
-    Spec = swagger_server:child_spec(swagger_ext, #{
-                                       ip => ListenAddress,
-                                       port => Port,
-                                       net_opts => [],
-                                       logic_handler => aehttp_dispatch_ext
-                                      }),
-    {ok, _} = supervisor:start_child(aehttp_sup, Spec),
-    ok.
 
-start_swagger_internal() ->
-    Port = get_internal_port(),
-    ListenAddress = get_internal_listen_address(),
-    Spec = swagger_server:child_spec(swagger_int, #{
-                                       ip => ListenAddress,
-                                       port => Port,
-                                       net_opts => [],
-                                       logic_handler => aehttp_dispatch_int
-                                      }),
-    {ok, _} = supervisor:start_child(aehttp_sup, Spec),
+start_http_api() ->
+    ok = start_http_api(external, aehttp_dispatch_ext, swagger_external_handler),
+    ok = start_http_api(internal, aehttp_dispatch_int, swagger_internal_handler).
+
+start_http_api(Target, LogicHandler, SwaggerHandler) ->
+    PoolSize = get_http_api_acceptors(Target),
+    Port = get_http_api_port(Target),
+    ListenAddress = get_http_api_listen_address(Target),
+
+    [{'_', Paths0}] = swagger_router:get_paths(LogicHandler),
```

Comment:
do we need to generate router code ourselfs as well?

Reply:
there is 'trails' library for that, but I am not sure if we should start with it

Reply:
Yes, all the request to `swagger_*` modules should be replaced by our generators